src/Main.py

```diff
@@ -2,65 +2,47 @@
 import streamlit_ext as ste
 import os
 import openai
-
 from doc_utils import extract_text_from_upload
 from templates import generate_latex, template_commands
 from prompt_engineering import generate_json_resume, tailor_resume
 from render import render_latex
 import json
 
-if __name__ == '__main__':
-
+def main():
     st.set_page_config(
         page_title="Resume2website",
         page_icon=":pen:",
         layout="wide",
         initial_sidebar_state="auto",
     )
 
-    st.markdown(
-        f"""
-        # Resume2Website
-        """,
-        unsafe_allow_html=True,
-    )
-
-    st.markdown(
-        "Welcome to Resume2Website! Drop your previous CV below, select one of the templates, and let the LLMs generate your resume for you"
-    )
-
+    st.markdown("# Resume2Website", unsafe_allow_html=True)
+    st.markdown("Welcome to Resume2Website! Drop your previous CV below, select one of the templates, and let the LLMs generate your resume for you")
 
     uploaded_file = st.file_uploader("Choose a file", type=["pdf", "docx", "txt", "json"])
-
     template_options = list(template_commands.keys())
 
-    if uploaded_file is not None:
-        # Get the CV data that we need to convert to json
+    if uploaded_file:
         text = extract_text_from_upload(uploaded_file)
-
         if len(text) < 50:
             st.warning("The text extracted from the uploaded file is too short. Are you sure this is the correct file?", icon="⚠️")
+            return
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rohankokkulabito/resfolio/pull/2#issuecomment-2225210660>#c0d9f6</a><span> - 07/12/2024, 09:44 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The early return statement 'return' inside the 'if len(text) < 50:' block will exit the 'main()' function but not the Streamlit app, potentially leaving the app in an inconsistent state.
</div>
<div id="fix">
<b>Fix:</b> Instead of using 'return', use 'st.stop()' to properly halt the execution of the Streamlit script, ensuring the app remains in a consistent state.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -29,0 +29,1 @@
 +            st.stop()
```
</div>
</details>
</div>
<br>


</div>